File: BackEnd/ContactsAPI/ContactsAPI/Services/IContactService.cs
using ContactsAPI.Models;
using Microsoft.AspNetCore.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ContactsAPI.Services
{
    public interface IContactService
    {
        public IActionResult GetContacts(string searchQuery, int page, int limit, Boolean isFavoriteSelected, Boolean sortAscending);
        public Contact GetContact(int id);
        public string AddContact(Contact value);
        public string UpdateContact(int id, Contact value);
        public string DeleteContact(int value);
    }
}

File: contacts-app/src/interfaces/IContactService.ts
import { TContactInformation, TContactInformationPayload, TContactListResult } from "../types/TContacts";

export default interface IContactService {
    getContacts(searchQuery: string, page: number, limit: number, isFavoriteSelected: boolean, sortAscending: boolean ): Promise<TContactListResult>;
    getContact(id: number): Promise<TContactInformation>;
    addContact(contactInfo: TContactInformationPayload): Promise<string>;
    updateContact(id: number, contactInfo: TContactInformationPayload): Promise<string>;
    deleteContact(id: number): Promise<string>;
}